src/summary/documentation/vital-card-level-two.component.tsx

```diff
@@ -0,0 +1,185 @@
+import React from "react";
+import { match } from "react-router";
+import { performPatientsVitalsSearch } from "./vitals-card.resource";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import { formatDate } from "./dimension-helpers";
+import styles from "./vital-card-level-two.css";
+import SummaryCard from "../cards/summary-card.component";
+import { useCurrentPatient } from "@openmrs/esm-api";
+import { Trans } from "react-i18next";
+
+export default function VitalsLevelTwo(props: VitalsLevelTwoProps) {
+  const resultsPerPage = 15;
+
+  const [patientVitals, setPatientVitals] = React.useState(null);
+  const [totalPages, setTotalPages] = React.useState(1);
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [showNextButton, setShowNextButton] = React.useState(false);
+  const [showPreviousButton, setShowPreviousButton] = React.useState(false);
+  const [currentPageResults, setCurrentPageResults] = React.useState([]);
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+
+  React.useEffect(() => {
+    if (!isLoadingPatient && patient) {
+      const subscription = performPatientsVitalsSearch(patient.id).subscribe(
+        vitals => {
+          setPatientVitals(vitals);
+          setTotalPages(Math.ceil(vitals.length / resultsPerPage));
+          setCurrentPageResults(vitals.slice(0, resultsPerPage));
+        },
+        createErrorHandler()
+      );
+
+      return () => subscription.unsubscribe();
+    }
+  }, [isLoadingPatient, patient]);
+
+  React.useEffect(() => {
+    {
+      patientVitals && currentPage * resultsPerPage > patientVitals.length
+        ? setShowNextButton(false)
+        : setShowNextButton(true);
+      currentPage !== 1
+        ? setShowPreviousButton(true)
+        : setShowPreviousButton(false);
+    }
+  }, [currentPageResults, currentPage, patientVitals]);
+
+  const nextPage = () => {
+    let upperBound = currentPage * resultsPerPage + resultsPerPage;
+    const lowerBound = currentPage * resultsPerPage;
+    if (upperBound > patientVitals.length) {
+      upperBound = patientVitals.length;
+    }
+    const pageResults = patientVitals.slice(lowerBound, upperBound);
+    setCurrentPageResults(pageResults);
+    setCurrentPage(currentPage + 1);
+  };
+
+  const previousPage = () => {
+    const lowerBound = resultsPerPage * (currentPage - 2);
+    const upperBound = resultsPerPage * (currentPage - 1);
+    const pageResults = patientVitals.slice(lowerBound, upperBound);
+    setCurrentPageResults(pageResults);
+    setCurrentPage(currentPage - 1);
+  };
+
+  function displayPatientsVitals() {
+    return (
+      <SummaryCard name="Vitals" match={props.match} styles={{ width: "100%" }}>
```

Comment:
The label "Vitals" also needs to be translated. Do you know if this is possible with our current <Trans> pattern?

Reply:
The following  **should not** be included/worried about in this PR. We can handle in another PR. 

I was playing around with this (passing a variable as a property to a component which then creates a Trans of it) on the login module. 

`
export default function Foo(props: FooProps) {

    return (
        <div>
            <Trans i18nKey={props.title}>{props.title}</Trans>            
        </div>
    )
}
`
then in login.component.tsx:

            <Foo title="hello"></Foo>
            <Foo title="bye"></Foo>
            <Foo title="hopefully"></Foo>
            <Foo title="yes"></Foo>

It seems that the babel plugin (to automatically add the key) doesn't recognize the variable when sent this way and does not automatically create a key in en.json. You can add the key manually into en.json but I noticed that the babel plugin will removes the manually inserted keys (probably because it removes anything that's not named in the current file). As a solution, we could set the babel plugin config for  discardOldKeys to false (at least for those modules in which you have the need to pass the key as a prop). Of course there are probably downsides to this with regards to key management but I think still worth it.

Reply:
Removed the translation part and will address it in a different PR